data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/AbstractJsonResultExtension.java

```diff
@@ -43,41 +39,27 @@ public void serialize(ResultExtension<T, ? extends Result> resultExtension, Json
         jsonGenerator.writeEndObject();
     }
 
-    @Override
-    public ResultExtension<T, ? extends Result> deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
-
-        ResultExtension<T, ? extends Result> resultExtension = new ResultExtension<>();
-
+    protected void deserializeResultExtension(JsonParser jsonParser, S resultExtension) throws IOException {
         while (!jsonParser.nextToken().isStructEnd()) {
-            switch (jsonParser.getCurrentName()) {
-                case "resultsPerVariant":
-                    jsonParser.nextToken();
-                    resultExtension.setResultMap(jsonParser.readValueAs(new TypeReference<Map<String, ? extends Result>>() {
-                    }));
-                    break;
-
-                default:
-                    throw new FaraoException("Unexpected field: " + jsonParser.getCurrentName());
+            if (jsonParser.getCurrentName().equals("resultsPerVariant")) {
+                jsonParser.nextToken();
+                resultExtension.setResultMap(jsonParser.readValueAs(new TypeReference<Map<String, ? extends Result>>() {
```

Comment:
I'm not sure about the wildcard here, we will see with remedial actions if it does still work

Reply:
I would almost define it like that `AbstractJsonResultExtension<T extends Identifiable<T>, R extends Result, S extends AbstractResultExtension<T, R>>` but I'm not sure it is useful yet